modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2MultiAttributeUserFilterTestCase.java

```diff
@@ -0,0 +1,396 @@
+        validateMultiAttributeFilteredUser(query.toString(), userIds.subList(1, 4), USERNAMES.subList(1, 4));
+    }
+
+    private void validateMultiAttributeFilteredUser(String filter, List<String> userIds, List<String> userNames)
+            throws IOException {
+
+        String userResourcePath = getUserPath() + "?filter=" + filter;
+        HttpGet request = new HttpGet(userResourcePath);
+        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        request.addHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 200, "User " +
+                "has not been retrieved successfully");
+
+        Object responseObj = JSONValue.parse(EntityUtils.toString(response.getEntity()));
+        EntityUtils.consume(response.getEntity());
+
+        int resultCount = (((JSONArray) ((JSONObject) responseObj).get("Resources")).size());
+        List<String> usernamesFromResponse = new ArrayList<>();
+        List<String> userIDsFromResponse = new ArrayList<>();
+        for (int i = 0; i < resultCount; i++) {
+            usernamesFromResponse.add(((JSONObject) ((JSONArray) ((JSONObject) responseObj).get("Resources")).
+                    get(i)).get(USER_NAME_ATTRIBUTE).toString());
+
+            userIDsFromResponse.add(((JSONObject) ((JSONArray) ((JSONObject) responseObj).get("Resources")).
+                    get(i)).get(ID_ATTRIBUTE).toString());
+        }
+
+        assertEquals(resultCount, userIds.size());
+        assertEquals(usernamesFromResponse, userNames);
+        assertEquals(userIDsFromResponse, userIds);
+    }
+
+    @Test(dependsOnMethods = "testMultiAttributeFilterUser")
+    public void testDeleteUser() throws Exception {
+
+        for (String userId : userIds) {
+            deleteUser(userId);
+        }
+    }
+
+    private void deleteUser(String userId) throws IOException {
+
+        String userResourcePath = getUserPath() + "/" + userId;
+        HttpDelete request = new HttpDelete(userResourcePath);
+        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        request.addHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 204, "User " +
+                "has not been retrieved successfully");
+
+        EntityUtils.consume(response.getEntity());
+
+        userResourcePath = getUserPath() + "/" + userId;
+        HttpGet getRequest = new HttpGet(userResourcePath);
+        getRequest.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        getRequest.addHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+
+        response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 404, "User " +
+                "has not been deleted successfully");
```

Comment:
This line can be merged with the previous line. Seems like it does not exceed 120 characters.

Reply:
Fixed